internal/ingress/controller/controller.go

```diff
@@ -323,79 +313,3 @@ func (n *KongController) Stop() error {
 
 	return nil
 }
-
-// handleBasicAuthUpdates updates basic-auth password field in Kong whenever it is changed.
-//
-// Kong hashes basic-auth passwords in DB and API responses once created.
-// Due to this reason, one can't perform a 'diff' with them.
-// This function filters for basic-auth password changes and applies them
-// to Kong as they happen.
-func (n *KongController) handleBasicAuthUpdates(ctx context.Context, event Event) error {
```

Comment:
So it seems like we introduced a bug when we added in the Secret-based credentials.
This function should be triggered when basic-auth creds in a secret change.
The bug has not come up because it applies to basic-auth credentials and DB-mode of Kong, which is not a common situation (I wonder where the users are who prompted me to add this function, we had quite a few reports for this problem in the past).

Reply:
Tracking this here: https://github.com/Kong/kubernetes-ingress-controller/issues/865

Not a blocker for this PR.